inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp

```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
-    const auto rowFunc = type_dispatch<TL>(depth, cv_type_id{}, KT{}, nullptr);
+    std::vector<Unit> xmaps(outSz.width);
+    int  maxdif = 0;
 
-    GAPI_DbgAssert(rowFunc);
+    for (int w = 0; w < outSz.width; w++) {
+        Unit map = mapper.map(w);
+        xmaps[w] = map;
 
-    rowFunc(dst, src0, src1, alpha, mapsx, beta, inSz, outSz, lpi, length);
-}
+        int dif = map.index1 - map.index0;
+        if (dif > maxdif)
+            maxdif = dif;
+    }
 
-template<typename T, class Mapper, typename KT, typename LT>
-static inline void calcRowLinear(const cv::gapi::fluid::View& in,
-                                 cv::gapi::fluid::Buffer& out,
-                                 cv::gapi::fluid::Buffer& scratch) {
-    GAPI_DbgAssert(is_cv_type_in_list<LT>(out.meta().depth));
+    // This assertion is critical for our trick with chunk sizes:
+    // we would expand a chunk it is is smaller than maximal size
+    GAPI_Assert(inSz.width >= maxdif);
 
-    auto  inSz = in.meta().size;
-    auto outSz = out.meta().size;
+    // pack the input chunks positions and coefficients into scratch-buffer,
+    // along with the maximal size of chunk (note that chunk size may vary)
 
-    auto inY = in.y();
-    int length = out.length();
-    int outY = out.y();
-    int lpi = out.lpi();
-    GAPI_DbgAssert(outY + lpi <= outSz.height);
+    size_t scratch_bytes =               sizeof(int)
+                         + outSz.width * sizeof(index_type)
+                         + outSz.width * sizeof(alpha_type) * maxdif
+                         +  inSz.width * sizeof(alpha_type);
+    Size scratch_size{static_cast<int>(scratch_bytes), 1};
 
-    GAPI_DbgAssert(lpi <= 4);
+    cv::GMatDesc desc;
+    desc.chan = 1;
+    desc.depth = CV_8UC1;
+    desc.size = scratch_size;
 
-    linearScratchDesc<T, Mapper, 1> scr(inSz.width, inSz.height, outSz.width,
-                                        outSz.height, scratch.OutLineB());
+    cv::gapi::fluid::Buffer buffer(desc);
+    scratch = std::move(buffer);
 
-    const auto* alpha = scr.alpha;
-    const auto* clone = scr.clone;
-    const auto* mapsx = scr.mapsx;
-    const auto* beta0 = scr.beta;
-    const auto* mapsy = scr.mapsy;
-    auto* tmp = scr.tmp;
+    auto *maxdf =  scratch.OutLine<int>();
+    auto *index = reinterpret_cast<index_type*>(maxdf + 1);
+    auto *alpha = reinterpret_cast<alpha_type*>(index + outSz.width);
+//  auto *vbuf  = reinterpret_cast<work_type *>(alpha + outSz.width * maxdif);
 
-    const auto* beta = beta0 + outY;
-    const T* src0[4];
-    const T* src1[4];
-    T* dst[4];
+    for (int w = 0; w < outSz.width; w++) {
+        // adjust input indices so that:
+        // - data chunk is exactly maxdif pixels
+        // - data chunk fits inside input width
+        int index0 = xmaps[w].index0;
+        int index1 = xmaps[w].index1;
+        int i0 = index0, i1 = index1;
+        i1 = (std::min)(i0 + maxdif, in.size.width);
```

Comment:
How does it work? Can it be `std::min(i0 + maxdif, in.size.width);` instead?

Reply:
Applied